h/views/account_signup.py

```diff
@@ -343,12 +349,47 @@ def encode_idinfo_token(
     provider_unique_id: str,
     issuer: JWTIssuer,
     audience: JWTAudience,
+    next_url: str,
 ):
     return {
         "idinfo": jwt_service.encode_symmetric(
-            IDInfo(provider_unique_id),
+            IDInfo(provider_unique_id, next_url=next_url),
```

Comment:
This encodes the `next` query param into the `idinfo` token query param before redirecting to the `/signup/<PROVIDER>?idinfo=token` page.